mysite/ShareBox/views.py

```diff
@@ -1,7 +1,56 @@
-from django.http import HttpResponse
-import requests
 import httplib
+import json
+from django.http import HttpResponse
+from ShareBox.models import Snippet
+from django.views.decorators.csrf import csrf_exempt
+from django.views.decorators.http import require_http_methods
+
+
+@require_http_methods(["POST"])
+@csrf_exempt
+def insert_db(request):
+    output={}
+    body_unicode = request.body.decode('utf-8')
+    body = json.loads(body_unicode)
+    name = body['snippet_name']
+    privacy = body['snippet_privacy']
+    expiry = body['snippet_expiry']
+    snippet_data = body['snippet_file']
+    new_snippet_box, created = Snippet.objects.get_or_create(
+          snippet_name=name, snippet_privacy=privacy, snippet_expiry=expiry,
+          snippet_file=snippet_data)
+    new_snippet_box.save()
+    url = "http://www/shareboxes.herokuapp.com/snippet/" + \
+            str(new_snippet_box.id) + "/"
+    output["url"] = url
+    return HttpResponse(json.dumps(output),
+                        content_type="application/json", status=201)
 
 
-def send_data(request, snippet_data):
-    pass
+@require_http_methods(["GET", "DELETE"])
+@csrf_exempt
+def query_db(request, input_id):
```

Comment:
Does this function's name make any sense? The function is called query_db. But it also deletes an entry. Can you figure out a way to call different functions for different HTTP methods?